Code/GraphMol/MolOps.cpp

```diff
@@ -660,21 +660,21 @@ std::vector<std::unique_ptr<MolSanitizeException>> detectChemistryProblems(
   return res;
 }
 
-std::vector<ROMOL_SPTR> getMolFrags(const ROMol &mol, bool sanitizeFrags,
-                                    INT_VECT *frags,
-                                    VECT_INT_VECT *fragsMolAtomMapping,
-                                    bool copyConformers) {
+namespace {
+std::vector<ROMol *> getRawFrags(const ROMol &mol, bool sanitizeFrags,
```

Comment:
A problem I see here is that we're going to leak memory if anything goes wrong in `getRawFrags()`

I think it's safer to have a base function that returns a vector of unique_ptrs and then have the function that needs to return shared_ptrs construct those from the unique_ptrs.

Reply:
I did wonder/worry about that situation.  However, the original code creates a bare molecule and then wraps it in a smart pointer so some of those concerns already existed.  I guess there's no reason why we can't make the code better along the way, though, so I'll do as you suggest.

Reply:
> I did wonder/worry about that situation. However, the original code creates a bare molecule and then wraps it in a smart pointer so some of those concerns already existed. I guess there's no reason why we can't make the code better along the way, though, so I'll do as you suggest.

The difference is that the original code creates the new molecule and then immediately wraps it in a shared ptr. Unless I missed something (which is entirely possible), I think there are a bunch more cases where the new code can leak

Reply:
No, I definitely made things worse from that perspective.  All tightened up now, I think.